pkg/nuctl/test/project_test.go

```diff
@@ -147,11 +152,215 @@ func (suite *projectGetTestSuite) TestDeleteWithFunctions() {
+
+	// reset output buffer for reading the nex output cleanly
+	suite.outputBuffer.Reset()
+	err := suite.ExecuteNuctlAndWait([]string{"get", "function", functionName}, map[string]string{
+		"output": "yaml",
+	}, false)
+	suite.Require().NoError(err)
+
+	function := functionconfig.Config{}
+	err = yaml.Unmarshal(suite.outputBuffer.Bytes(), &function)
+	suite.Require().NoError(err)
+
+	suite.Assert().Equal(functionName, function.Meta.Name)
+	if imported {
+
+		// get imported functions
+		err = suite.ExecuteNuctl([]string{"get", "function", functionName}, nil)
+		suite.Require().NoError(err)
+
+		// ensure function state is imported
+		suite.findPatternsInOutput([]string{"imported"}, nil)
+	}
+}
+
+func (suite *projectExportImportTestSuite) assertFunctionEventExistenceByFunction(functionEventDisplayName,
+	functionName string) string {
+
+	// reset output buffer for reading the nex output cleanly
+	suite.outputBuffer.Reset()
+	err := suite.ExecuteNuctlAndWait([]string{"get", "functionevent"}, map[string]string{
+		"output": "yaml",
+		"function": functionName,
+	}, false)
+	suite.Require().NoError(err)
+
+	functionEvent := platform.FunctionEventConfig{}
+	err = yaml.Unmarshal(suite.outputBuffer.Bytes(), &functionEvent)
+	suite.Require().NoError(err)
+
+	suite.Assert().Equal(functionEventDisplayName, functionEvent.Spec.DisplayName)
+	suite.Assert().Equal(functionName, functionEvent.Meta.Labels["nuclio.io/function-name"])
+	return functionEvent.Meta.Name
+}
+
+func (suite *projectExportImportTestSuite) TestExportProject() {
+	projectName := "test-project"
+	functionName := "test-function"
+	functionEventName := "test-function-event"
+
+	suite.createProject(projectName)
+	suite.createFunction(functionName, projectName)
+	suite.createFunctionEvent(functionEventName, functionName)
+
+	defer suite.ExecuteNuctl([]string{"delete", "proj", projectName}, nil)
+	defer suite.ExecuteNuctl([]string{"delete", "fu", functionName}, nil)
+	defer suite.ExecuteNuctl([]string{"delete", "fe", functionEventName}, nil)
+
+	// reset output buffer for reading the nex output cleanly
+	suite.outputBuffer.Reset()
+
+	// export the project
+	err := suite.ExecuteNuctlAndWait([]string{"export", "proj", projectName, "--verbose"}, map[string]string{}, false)
+	suite.Require().NoError(err)
+
+	exportedProjectConfig := &command.ProjectImportConfig{}
+	err = yaml.Unmarshal(suite.outputBuffer.Bytes(), &exportedProjectConfig)
+	suite.Require().NoError(err)
+
+	suite.Assert().Equal(exportedProjectConfig.Project.Meta.Name, projectName)
+	suite.Assert().Equal(exportedProjectConfig.Functions[functionName].Meta.Name, functionName)
+	suite.Assert().Equal(exportedProjectConfig.FunctionEvents[functionEventName].Meta.Name, functionEventName)
+}
+
+func (suite *projectExportImportTestSuite) TestImportProject() {
+	projectConfigPath := path.Join(suite.GetImportsDir(), "project.yaml")
+	projectName := "test-project"
```

Comment:
random suffix as well

Reply:
Not sure it's possible here, as it's written in the yaml.
The only other way is to open the yaml and save a temporary one with the random suffix, wdyt?

Reply:
You can create a project instance, fill it with random stuff, create its file onto >tmp and do it.

The reason for doing it is not to maintain that file if there will be fields changes, you will find about it during tuneimte rather during compilation time